src/System.Numerics.Vectors/ref/System.Numerics.Vectors.cs

```diff
@@ -52,6 +52,27 @@ public partial struct Matrix3x2 : System.IEquatable<System.Numerics.Matrix3x2>
         public static System.Numerics.Matrix3x2 operator -(System.Numerics.Matrix3x2 value) { throw null; }
         public static System.Numerics.Matrix3x2 Subtract(System.Numerics.Matrix3x2 value1, System.Numerics.Matrix3x2 value2) { throw null; }
         public override string ToString() { throw null; }
+#if FEATURE_REF_OVERLOADS
```

Comment:
Why is this needed?

Reply:
Because we build for netfx, and these APIs aren't in desktop (yet).  But we wanted to make these APIs available in as many places as possible.  So we are starting with netcoreapp and uap.

Reply:
@weshaggard, in many projects we add to the refs without special-casing them for netfx.  What's special about this one and others that still require special-casing?

@eerhardt, personally I'd prefer if these were added in a .netcoreapp.cs specific file rather than using #ifdefs, but I'm ok being overruled if this is the practice we've already established elsewhere.

Reply:
The reason I didn’t put it in .netcoreapp.cs file was because it gets compiled in for both netcoreapp AND uap. When searching other scenarios, I found the FEATURE pattern being used.

Ex https://github.com/dotnet/corefx/blob/1ce37dfb5a03e5decdb25ee67ef765f89424ceb1/src/System.Security.Cryptography.Cng/ref/System.Security.Cryptography.Cng.cs

Reply:
> The reason I didn’t put it in .netcoreapp.cs file was because it gets compiled in for both netcoreapp AND uap.

We use "netcoreapp" all over the place when it's also used for uap.  That's now the case for the vast majority of new APIs across corefx.  The terminology needs to be cleaned up.

Reply:
> We use "netcoreapp" all over the place when it's also used for uap.

This seems like an anti-pattern to me.  There are plenty of places where `.netcoreapp.cs` means "only used on netcoreapp".  Here's just a couple I found really quickly:

https://github.com/dotnet/corefx/blob/6f08f5a12a4723a15efbbd7ba9bc0bcc929c41a6/src/System.Reflection.Metadata/src/System.Reflection.Metadata.csproj#L106
https://github.com/dotnet/corefx/blob/6f08f5a12a4723a15efbbd7ba9bc0bcc929c41a6/src/System.Threading.Tasks/tests/System.Threading.Tasks.Tests.csproj#L55-L58
https://github.com/dotnet/corefx/blob/6f08f5a12a4723a15efbbd7ba9bc0bcc929c41a6/src/System.Drawing.Primitives/tests/System.Drawing.Primitives.Tests.csproj#L25-L29

----

This one even mixes them both into the same project:

https://github.com/dotnet/corefx/blob/6f08f5a12a4723a15efbbd7ba9bc0bcc929c41a6/src/System.Runtime.Extensions/tests/System.Runtime.Extensions.Tests.csproj#L34-L39

----

I assume the intention of suffixing these files with a TFM was so that it was easily recognizable that this file was only compiled for this TFM.  When we use one TFM to mean it is sometimes compiled for 2 TFMs, the only way you can know is to go look in the .csproj file.

I'd be happy to split these out into a separate file, but maybe a `.netcoreapp.uap.cs` suffix would be better?

Reply:
Like I said, the terminology needs updating. "netcoreapp" has morphed into also including all new APIs, as all new APIs are being added for all platforms that use corefx/coreclr/corert.